### PR TITLE
WIP: Creating how to, "How to install TiddlyWiki5 on Heroku"

### PR DESCRIPTION
**Pull request by jayfresh**
Still needs formatting sorting

### CODE REVIEW

**Comment by danielo515** [comment]
Dear @jayfresh 

Have you ended creating the instructions to set up this with couch db?

Many thanks

**Comment by jayfresh** [comment]
I didn't write any more, but I did have a productive conversation[1] with @wshallum which ended with an update to his CouchDB Adaptor[2], making it work in the cross-domain setup required here.

[1] https://github.com/wshallum/couchadaptor/pull/2
[2] https://github.com/wshallum/couchadaptor

**Comment by danielo515** [comment]
Hello @jayfresh 
I was aware about your suggestions for the couch adaptor. Thank you very much. Now I'm using it with the single file edition, and I also made a couple of pull requests. Thanks

**Comment by jayfresh** [comment]
Where are the pull requests?

**Comment by danielo515** [comment]
On the couch adaptor repo. Sorry If I was unclear about that. It is for fixing the problem with dates and other specific TW fields.

**Comment by bradoyler** [comment]
@jayfresh How does one change the datastore to couchdb?

**Comment by sukima** [comment]
Does anyone know if there is a redis saver for TiddlyWiki? Heroku is pretty good about support for redis-to-go. Maybe it is worth making a redis saver plugin.

I mused about rewriting the server code in TiddlyWiki to match the [PHP TiddlySpot server](http://tiddlywiki.com/#Saving%20on%20a%20PHP%20Server) so it would be implemented in Node.js and not PHP. But I got confused at the API and quickly lost motivation. But even if that did come to fruition it wouldn't help the filesystem issue but it would provide a better framework for choosing the backend that supports the server which hosts the application.

**Comment by tobibeer** [comment]
> Does anyone know if there is a redis saver for TiddlyWiki?

I'd be surprised to find there was.

Suppose we don't know **what.redis.is** rather than  that we're already busy leveraging **redis.io**...

What would be your idea behind CRUDing tiddlers to an "in-memory data structure store"? Is it like some intermittent non-connecitvity storage only to later sync the load to a cloud backend like Heroku? Or would www.redis.io and then www.redistogo.com be some server-side "in-memory data structure store" and so the question becomes: Why have some middleware that doesn't persist data quite the way you eventually want them to?

> I mused about rewriting the server code in TiddlyWiki to match the PHP TiddlySpot server so it would be implemented in Node.js and not PHP.

*Mused* as in *started to port the TiddlySpot framework* to some node.js server ...other than the one being used by TiddlyWiki? Besides basic saving, you'd want registration, authentication, and then a "TiddlyNode" controlpanel, so as to be able to eventually delete a TiddlyNode. I'd have an idea of how to go about that in php, but not in node.

> But I got confused at the API and quickly lost motivation.

Which? Where? Or are you actually talking about [TiddlyWeb](http://tiddlyweb.com) / [TiddlySpace](http://tiddlyspace.com)?

File: editions/tw5.com/tiddlers/howtos/How to install TiddlyWiki5 on Heroku.tid [hunk 1]
```diff
@@ -0,0 +1,49 @@
+created: 20141010110448509
+modified: 20141010110448509
+title: How to install TiddlyWiki5 on Heroku
+tags: [[TiddlyWiki on Node.js]]
+type: text/vnd.tiddlywiki
+
+1. Creating the local development environment
+
+npm install -g tiddlywiki # install tw5 globally
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/howtos/How to install TiddlyWiki5 on Heroku.tid]: These command line steps should be in a code block. Also the docs refer to "TiddlyWiki", not "tw5".

File: editions/tw5.com/tiddlers/howtos/How to install TiddlyWiki5 on Heroku.tid [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+3. Create the heroku application
+
+I did created a new app on the heroku dashboard, and then followed the instructions to set it up locally. NB: you need heroku toolbelt installed locally.
+
+cd newwikifolder # move into your new wiki directory
+heroku git:remote -a newapp # add the heroku git remote, where 'newapp' is the name of the app you created
+git push heroku master # push the master branch to heroku, which launches tw5
+
+Now TW5 will be running at http://<newapp>.herokuapp.com.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/howtos/How to install TiddlyWiki5 on Heroku.tid]: Probably work wrapping the URL in backticks

**Comment by Jermolene** [comment]
Hi @jayfresh apologies for the delay (!). Happy to merge this if the documentation style issues I've noted can be addressed, and we can verify that the instructions still work

**Comment by pmario** [comment]
@jayfresh ... Is this PR still a thing that works?

**Comment by jayfresh** [comment]
I'm not sure Mario, but please feel free to update here if you try it out
and it works!

**Comment by pmario** [comment]
@Jermolene -- User does not show any GH activity since 2020 and Heroku does not have a free tier anymore, one has to have a paid account to test the info. 

I think we should close this PR and wait for updated info, which uses the Herokup platform as it is now.

**Comment by Jermolene** [comment]
Thanks @pmario

**Pull request closed without merging**